### PR TITLE
refactor: process push remove event - WPB-10199

### PR DESCRIPTION
**Pull request by jullianm**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10199" title="WPB-10199" target="_blank">TaskWPB-10199</a>  [iOS] Process UserEvent.pushRemove
  </td></table>
  <br />
 

### Key points

This PR is part of the quick sync refactoring plan and is related to processing the multiple events we receive from the backend or the push channel.

Specifically, this PR is about porting the existing implementation of the `UserPushRemove` event.

### Testing

UTs cover the following use cases, ensuring that
- push token is properly removed from user defaults

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor/UserPushRemoveEventProcessor.swift
Comment (netbe) [inline WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor/UserPushRemoveEventProcessor.swift]: does that makes sense to clean the signature then:
```suggestion
    func processEvent() {
```

File: WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift
Comment (netbe) [inline WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift]: can we take care of the TODO before merging?
also it would be better to use a constant for the key, in case we have to clean up it's easier to look for.

**Review by netbe — COMMENTED** [review]
left 2 comments before approving

File: WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift
Comment (johnxnguyen) [inline WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift]: Perhaps we should use a temporary defaults to avoid polluting the real one.

File: WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift
Comment (johnxnguyen) [inline WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift]: Shouldn't we test with a mock user repository?

File: WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor/UserPushRemoveEventProcessor.swift
Comment (jullianm) [reply WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor/UserPushRemoveEventProcessor.swift]: fixed thx

File: WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift
Comment (jullianm) [reply WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift]: yes this one is related to this PR [here](https://github.com/wireapp/wire-ios/pull/1932) that includes a **scoped** UserDefaults approach and a propertyWrapper. So keeping this one on hold for now

File: WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift
Comment (jullianm) [reply WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift]: in theory yes but in practice processors are mostly just passthrough objects to the repository (they don't do much except forwarding data to the repo). 
So if we pass mock repository to processors, what we can test is that repo methods are correctly invocated.

It makes me wonder if we should even test such processors when repositories are already tested at the WireDomain level and they perform most of the work.

Comment (jullianm) [reply WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift]: resolving this conversation after aligning on how we test processors that is: injecting the mock repository and ensuring that repo methods are correctly invocated.

File: WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift
Comment (jullianm) [reply WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift]: done in https://github.com/wireapp/wire-ios/pull/1948/commits/09836061f1870849edc69be3d204a5c2afb69df8

File: WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift
Comment (jullianm) [reply WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift]: done in https://github.com/wireapp/wire-ios/pull/1948/commits/09836061f1870849edc69be3d204a5c2afb69df8

File: WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift
Comment (netbe) [inline WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift]:
```suggestion
    /// Removes user push token from storage.
```
or 
```suggestion
    /// Removes user push token from user defaults.
```

I guess we don't want to share implementation details in public interface

**Review by netbe — APPROVED** [review]
Looks good, just added some suggestion about documentation comment

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**
